lib/elixir/lib/option_parser.ex

```diff
@@ -112,6 +112,8 @@ defmodule OptionParser do
 
   ### Parsing dynamic switches
 
+  WARNING: this option is deprecated.
+
```

Comment:
Maybe if the option is deprecated we should delete it from the doc?

Reply:
I saw it's done like that for `Behaviour`, should we just remove all this @josevalim?